See https://github.com/gap-system/gap/pull/3294

commit c3a9f302c3a8f3bbd682e7517821aa3deaa03f13
Author: Max Horn <max@quendi.de>
Date:   Fri Feb 22 22:05:57 2019 +0100

    buildsys: make some minor progress on 'make install'
    
    Specifically, re-enable install-libgap and install-headers, improve them, and
    add more install-FOO subtargets.
    
    However, we deliberately keep printing an error in install target, and all
    other subtargets print warnings.
    
    This way, packagers who dig can potentially benefit from these subtargets; but
    if they complain about them, we can point at the warnings, the fact that the
    targets are undocumented, and the TODO comments in them.
    
    Also, people who want to help with `make install` support get at least some
    hints about the issues need to be resolved.

diff --git a/Makefile.rules b/Makefile.rules
index 16a05eb..c3a2b05 100644
--- a/Makefile.rules
+++ b/Makefile.rules
@@ -260,7 +260,6 @@ QUIET_CC      = @echo "   C       $< => $@";
 QUIET_CXX     = @echo "   CXX     $< => $@";
 QUIET_LINK    = @echo "   LINK    $@";
 QUIET_SED     = @echo "   SED     $< => $@";
-QUIET_INSTALL = @echo "   INSTALL $< => $@";
 QUIET         = @
 LIBTOOL 	 += --silent
 endif
@@ -477,26 +476,68 @@ clean:
 
 LTINSTALL=$(LIBTOOL) --mode=install $(INSTALL)
 
-install: install-libgap install-headers
+install:
 	@echo "Error, 'make install' has not yet been implemented"
 	exit 1
-# 	$(QUIET_INSTALL)$(INSTALL) -d -m 0755 $(DESTDIR)$(bindir)
-# 	$(QUIET_INSTALL)$(INSTALL) gap $(DESTDIR)$(bindir)
-# 	$(QUIET_INSTALL)$(INSTALL) gac $(DESTDIR)$(bindir)
-# 
-# install-headers:
-# 	$(QUIET_INSTALL)$(INSTALL) -d -m 0755 $(DESTDIR)$(includedir)/gap
-# 	$(QUIET_INSTALL)$(INSTALL) -d -m 0755 $(DESTDIR)$(includedir)/gap/hpc
-# 	$(QUIET_INSTALL)$(INSTALL) -m 0644 $(srcdir)/src/*.h $(DESTDIR)$(includedir)/gap
-# 	$(QUIET_INSTALL)$(INSTALL) -m 0644 $(srcdir)/src/hpc/*.h $(DESTDIR)$(includedir)/gap/hpc
-# 	# TODO: take care of config.h
-# 
-# install-libgap: libgap.la
-# 	$(QUIET_INSTALL)$(INSTALL) -d -m 0755 $(DESTDIR)$(libdir)
-# 	$(QUIET_INSTALL)$(LTINSTALL) libgap.la $(DESTDIR)$(libdir)
-
 
-.PHONY: install install-libgap
+install-bin:
+	@echo "Warning, 'make install-bin' is incomplete"
+	$(INSTALL) -d -m 0755 $(DESTDIR)$(bindir)
+	$(INSTALL) -m 0755 gap $(DESTDIR)$(bindir)/gap.real
+	echo "!/bin/sh\nexec \"$(bindir)/gap.real\" -l \"$(datarootdir)/gap\" \"$@\"" > gap-wrapper.sh
+	$(INSTALL) -m 0755 gap-wrapper.sh $(DESTDIR)$(bindir)/gap
+	# TODO: make gac installable; this requires adjusting path in it, and
+	# installing the libtool script generated by configure somewhere; and then
+	# putting that path to it into gac)
+	#$(INSTALL) gac $(DESTDIR)$(bindir)
+
+install-gaproot:
+	@echo "Warning, 'make install-gaproot' is incomplete"
+	$(INSTALL) -d -m 0755 $(DESTDIR)$(datarootdir)/gap
+	# TODO: update paths and FLAGS in sysinfo.gap
+	$(INSTALL) -m 0644 $(srcdir)/sysinfo.gap $(DESTDIR)$(datarootdir)/gap
+	# the following lines should not use `cp -r`, which is not quite portable,
+	# and which also may not deal with file permissions correctly
+	cp -r $(srcdir)/doc $(DESTDIR)$(datarootdir)/gap/  # FIXME: don't use `cp -r`
+	cp -r $(srcdir)/grp $(DESTDIR)$(datarootdir)/gap/  # FIXME: don't use `cp -r`
+	cp -r $(srcdir)/lib $(DESTDIR)$(datarootdir)/gap/  # FIXME: don't use `cp -r`
+	cp -r $(srcdir)/tst $(DESTDIR)$(datarootdir)/gap/  # FIXME: don't use `cp -r`
+	# TODO: what about CITATION, CONTRIBUTING.md, COPYRIGHT, INSTALL.md,
+	# LICENSE, README* ? Copy them also here? Or into some other path?
+	# TODO: also copy bin/BuildPackage.sh, as it is very useful?
+
+install-pkg:
+	@echo "Warning, 'make install-pkg' is incomplete"
+	# TODO: should we try to compiled packages? perhaps better not, leave it
+	# to the packagers, as they also need to consider build dependencies ?! If
+	# we do try it, then we may need to deal with adjusting paths and so on...
+	# we can't really install packages by any other means that copying their
+	# whole directory... One strategy to overcome this: in the future, we
+	# demand that packages which must be built using `make` also support
+	#     `make install prefix=DIR DESTDIR=dir`
+	# and/or allow packages to have a `Install(prefix, destdir)` field in
+	# their `PackageInfo.g` which contains a function that takes care of
+	# installation (e.g. by running `make install`, or by any other means).
+	# This could also be used by the PackageManager package (or perhaps we
+	# should just use PackageManager ourselves, and/or recommend it to
+	# packagers).
+
+install-headers:
+	@echo "Warning, 'make install-headers' is incomplete"
+	$(INSTALL) -d -m 0755 $(DESTDIR)$(includedir)/gap
+	$(INSTALL) -m 0644 $(srcdir)/src/*.h $(DESTDIR)$(includedir)/gap
+	# TODO: run the following only if HPC-GAP is enabled?
+	$(INSTALL) -d -m 0755 $(DESTDIR)$(includedir)/gap/hpc
+	$(INSTALL) -m 0644 $(srcdir)/src/hpc/*.h $(DESTDIR)$(includedir)/gap/hpc
+	# TODO: take care of config.h, this is difficult
+
+install-libgap: libgap.la
+	@echo "Warning, 'make install-libgap' is incomplete"
+	$(INSTALL) -d -m 0755 $(DESTDIR)$(libdir)
+	$(LTINSTALL) libgap.la $(DESTDIR)$(libdir)
+
+
+.PHONY: install install-bin install-gaproot install-headers install-libgap
 
 ########################################################################
 # Building subprojects
